File: 380-insert-delete-getrandom-o-1/randomized_set.py
import random 

class RandomizedSet:
    def __init__(self):
        self.map = {}
        self.array = []
    
    def insert(self, val: int) -> bool:
        if val in self.map:
            return False
        self.map[val] = len(self.array)
        self.array.append(val)

        return True
    
    def remove(self, val: int) -> bool:
        if val not in self.map:
            return False
        
        index = self.map[val]
        last_value = self.array[-1]

        self.array[index] = last_value
        self.map[last_value] = index

        self.array.pop()
        del self.map[val]

        return True
    
    def get_random(self) -> int:
        return random.choice(self.array)
    
# Example usage:
randomized_set = RandomizedSet()
print(randomized_set.insert(1))       # True
print(randomized_set.remove(2))       # False
print(randomized_set.insert(2))       # True
print(randomized_set.get_random())     # Either 1 or 2
print(randomized_set.remove(1))       # True
print(randomized_set.insert(2))       # False
print(randomized_set.get_random())     # 2

File: 380-insert-delete-getrandom-o-1/randomized_set.js
class RandomizedSet {
    constructor() {
        this.map = new Map();
        this.array = [];
    }

    insert(val) {
        if(this.map.has(val)) {
            return false;
        }

        this.map.set(val, this.array.length);
        this.array.push(val);
        return true;
    }

    remove(val) {
        if(!this.map.has(val)) {
            return false;
        }

        const index = this.map.get(val);
        const lastValue = this.array[this.array.length-1];

        this.array[index] = lastValue;
        this.map.set(lastValue, index);

        this.array.pop();
        this.map.delete(val);

        return true;
    }


    getRandom() {
        const randomIndex = Math.floor(Math.random() * this.array.length);

        return this.array[randomIndex];
    }
}

// Example usage:
const randomizedSet = new RandomizedSet();
console.log(randomizedSet.insert(1)); // true
console.log(randomizedSet.remove(2)); // false
console.log(randomizedSet.insert(2)); // true
console.log(randomizedSet.getRandom()); // Either 1 or 2
console.log(randomizedSet.remove(1)); // true
console.log(randomizedSet.insert(2)); // false
console.log(randomizedSet.getRandom()); // 2